simpleclient_spring_boot/src/main/java/io/prometheus/client/spring/EnablePrometheusCollectorRegistration.java

```diff
@@ -0,0 +1,52 @@
+package io.prometheus.client.spring;
+
+import org.springframework.context.annotation.Import;
+
+import java.lang.annotation.Documented;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Inherited;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * An {@literal @}Enable style annotation that provides CollectorRegistry related functions.
+ *
+ * <pre><code>
+ *{@literal @}Configuration
+ *{@literal @}EnablePrometheusCollectorRegistration(useDefault = true, register = true)
+ * public class MyConfiguration {
+ *
+ *  {@literal @}Bean
+ *   public Counter counter() {
```

Comment:
Apologies for being imprecise. 

> That's not a usual way of doing testing, that's going to break any singletons implemented as static final.

It is the usual way of testing Spring applications.
NB Spring can mark any object/bean as a singleton and has a registry, so it doesn't need to use a static final field reference to ensure a given object is accessible to other classes.

Static field singletons are somewhat notorious for causing memory leaks in containers with multiple classloaders, especially when they're used as a registry. 

If classes loaded in different classloaders are added to the static singleton, their classloader can't be garbage collected if it is otherwise ready to be dereferenced - unless the registry clears their reference, which it may not be able to do.

> Are you sure those are the semantics of how Spring does tests, as that sounds very fragile?

Yes, I'm sure. Perhaps it's just a side-effect of a conflict of styles.

> That sounds like a bug in Java.

Well... funny you should mention that, but even the double-checked-locking idiom was broken until Java 5. In this case, I don't think it's a bug in Java. Here's a good write up about singletons & safety: https://shipilev.net/blog/2014/safe-public-construction/#_singletons_and_singleton_factories